### PR TITLE
feat: added a flag --skip-preview to keploy cli to skip line by line coverage

### PR DESCRIPTION
**Pull request by debayangg**
## Related Issue
  - Info about Issue or bug

Closes: #[2033](https://github.com/keploy/keploy/issues/2033)
Associated docs PR for this feature: #[432](https://github.com/keploy/docs/pull/432)

#### Describe the changes you've made
Adds a new flag to keploy's test mode which skips line by line coverage but displays the final code coverage

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

New functionality:

https://github.com/user-attachments/assets/ba144cea-ed11-4879-b016-86621ad79806

### CODE REVIEW

**Comment by debayangg** [comment]
@shivamsouravjha pls review

**Comment by shivamsouravjha** [comment]
Pretty sure there's some issue with the implemenation~ 

Would suggest deleting the local nyc files to replicate this.

https://github.com/user-attachments/assets/ed991877-ffa4-41c4-bad9-7e0403afa2bb

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]
Also I believe the implementation is wrong 
image
you're removing nyc from command which would disable coverage overall.

**Comment by debayangg** [comment]
I tried to run skip-preview with your instructions and I did face an error after deleting the .nyc_output folder.

Since the line-by-line coverage preview is shown due to nyc, I changed the preprocess command to "nyc --clean=$CLEAN --reporter=none " after going through [link](https://istanbul.js.org/docs/advanced/alternative-reporters/) and it skipped the preview while also generating the .nyc_output folder which is needed for calculating final cov. Should I create a new preprocess_skippreview func for js or change the func to take an extra bool param for skipreview?

**Comment by shivamsouravjha** [comment]
you can put a small append and link it to flag

File: pkg/service/replay/replay.go [hunk 1]
```diff
@@ -164,8 +164,14 @@ func (r *Replayer) Start(ctx context.Context) error {
 
 			if err != nil {
 				r.config.Test.SkipCoverage = true
+			} else if r.config.Test.Language == models.Javascript && r.config.Test.SkipPreview {
+				r.config.Command = strings.Replace(r.config.Command, "nyc --clean=$CLEAN ", "nyc --clean=$CLEAN --reporter=none ", 1)
 			}
+
 		}
+	}
```
Comment (shivamsouravjha) [inline pkg/service/replay/replay.go:172]: instead of replacing you can use the pass a flag in preprocess and directly append there.

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]

File: pkg/service/replay/replay.go (hunk 1 above)
Comment (shivamsouravjha) [reply pkg/service/replay/replay.go:172]: yes something of that sort.

**Comment by shivamsouravjha** [comment]
also checked the keploy.yml file created it sets disableLineCoverage to false, ideally should be true.

File: pkg/service/replay/replay.go [hunk 2]
```diff
@@ -160,12 +160,16 @@ func (r *Replayer) Start(ctx context.Context) error {
...
 
 			if err != nil {
 				r.config.Test.SkipCoverage = true
 			}
+
 		}
+	}
+
+	if !r.config.Test.SkipCoverage {
 		err = os.Setenv("CLEAN", "true") // related to javascript coverage calculation
```
Comment (shivamsouravjha) [inline pkg/service/replay/replay.go:173]: why is this needed?

Comment (debayangg) [reply pkg/service/replay/replay.go:173]: I had done this change for my previous approach, currently it doesn't have any need. Should I revert to previous syntax?

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]
please address the comment about keploy.yml updated with disable preview being set to true.

**Comment by shivamsouravjha** [comment]
Issue is solved by other PR which was raised to solved a similar issue.
